src/runner/basic.rs

```diff
@@ -0,0 +1,673 @@
+//! Default [`Runner`] implementation.
+
+use std::{
+    cmp,
+    collections::HashMap,
+    fmt::{Debug, Formatter},
+    mem,
+    panic::{self, AssertUnwindSafe},
+    path::PathBuf,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use futures::{
+    channel::mpsc,
+    future::{self, Either, FutureExt as _},
+    lock::Mutex,
+    stream::{self, LocalBoxStream, Stream, StreamExt as _, TryStreamExt as _},
+    TryFutureExt,
+};
+use itertools::Itertools as _;
+use regex::Regex;
+
+use crate::{
+    event::{self, Info},
+    feature::FeatureExt as _,
+    step, Runner, Step, World,
+};
+
+/// Default [`Runner`] implementation.
+///
+/// Can execute [`Scenario`]s concurrently based on custom function, which
+/// returns [`ScenarioType`]. Also can limit maximum number of concurrent
+/// [`Scenario`]s.
+///
+/// [`Scenario`]: gherkin::Scenario
+pub struct Basic<World, F> {
+    max_concurrent_scenarios: Option<usize>,
+    steps: step::Collection<World>,
+    which_scenario: F,
```

Comment:
The purpose of this field and its type parameter is quite unclear here. Please, document it with a proper explanation.